gradoop-flink/src/test/java/org/gradoop/flink/algorithms/gelly/connectedcomponents/ConnectedComponentsTest.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright © 2014 - 2017 Leipzig University (Database Research Group)
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.algorithms.gelly.connectedcomponents;
+
+import org.apache.flink.api.java.DataSet;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.bool.And;
+import org.gradoop.flink.model.impl.functions.utils.EqualsByPropertyValue;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.junit.Test;
+
+public class ConnectedComponentsTest extends GradoopFlinkTestBase {
```

Comment:
If we had the suggested implementation that returns a `GraphCollection` the test could be replaced with something like what is done in [SplitTest](https://github.com/dbs-leipzig/gradoop/blob/master/gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/split/SplitTest.java): Just comparing the elements of the `GraphCollection` via [GraphCollection#equalsByGraphElementData](https://github.com/dbs-leipzig/gradoop/blob/master/gradoop-flink/src/main/java/org/gradoop/flink/model/api/epgm/GraphCollection.java#L349). Then we could also remove the one-use-only class `EqualsByPropertyValue`